src/CoreWCF.Primitives/src/CoreWCF/Description/IEndpointAddressForMetadataProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace CoreWCF.Description
+{
+    public interface IEndpointAddressForMetadataProvider
+    {
+        System.Uri GetEndpointAddress();
+    }
```

Comment:
I think we should make this more flexible and have it accept the HttpRequest (or HttpContext, but HttpRequest feels like the better match to me, but not a strong opinion on it) as a parameter. That way an implementation can do things like read the X-Forwarded... headers etc or do something more custom like return a different address for WAN and LAN clients.

Reply:
I updated the signature to accept the `HttpRequest`